### PR TITLE
[JENKINS-57557] Add integration test with github-branch-source

### PR DESCRIPTION
**Pull request by fcojfernandez**
See [JENKINS-57557](https://issues.jenkins-ci.org/browse/JENKINS-57557). _configuration-as-code_ is compatible with the last version of _github-branch-sources_. This PR adds an integration test to check possible future issues.

### Your checklist for this pull request

🚨 Please review the [guidelines for contributing](../blob/master/docs/CONTRIBUTING.md) to this repository.

- [X] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [X] Link to relevant issues in GitHub or in [Jenkins JIRA](https://issues.jenkins-ci.org)
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [X] Did you provide a test-case? That demonstrates feature works or fixes the issue.

@varyvol @MRamonLeon @alecharp @rsandell 
@timja @casz

### CODE REVIEW

File: plugin/src/main/java/io/jenkins/plugins/casc/impl/configurators/HeteroDescribableConfigurator.java [hunk 1]
```diff
@@ -203,14 +203,14 @@ public CNode describe(T instance, ConfigurationContext context) {
 
     private Option<Descriptor<T>> lookupDescriptor(String symbol, CNode config) {
         return getDescriptors()
-            .filter(descriptor -> findByPreferredSymbol(descriptor, symbol) || findBySymbols(descriptor, symbol, config))
```
Comment (casz) [inline plugin/src/main/java/io/jenkins/plugins/casc/impl/configurators/HeteroDescribableConfigurator.java]: Actually this is the correct formatting: https://github.com/jenkinsci/configuration-as-code-plugin/blob/master/.idea/codeStyles/Project.xml#L89

😆

Comment (casz) [reply plugin/src/main/java/io/jenkins/plugins/casc/impl/configurators/HeteroDescribableConfigurator.java]: Perhaps not, we should take the styling fix in a separate PR :)

**Comment by varyvol** [comment]
@fcojfernandez I can see also [github-branch-source-plugin #247](https://github.com/jenkinsci/github-branch-source-plugin/pull/247). Does it make sense to repeat the test in both plugins? Which is the usual strategy?

File: plugin/src/main/java/io/jenkins/plugins/casc/impl/configurators/HeteroDescribableConfigurator.java (hunk 1 above)
Comment (fcojfernandez) [reply plugin/src/main/java/io/jenkins/plugins/casc/impl/configurators/HeteroDescribableConfigurator.java]: The format change was made automatically by IntelliJ. My vote is always to make format changes in separate PRs, but as your preference.

**Comment by fcojfernandez** [comment]
@varyvol AFAIK, there is no strategy. I made the decision to add both tests since there was none. This way we can check:
* Future changes in _github-branch-source-plugin_ are still compatible at least with _configuration-as-code:1.30_
* Future changes in _configuration-as-code_ are still working with _github-branch-source-plugin_

**Comment by fcojfernandez** [comment]
Hi @timja @casz 

Is it missing something else from my side? Can this PR be merged?

**Comment by timja** [comment]
Meant to reply yesterday saying I'll merge it today, sorry. Done now

**Comment by fcojfernandez** [comment]
@timja thanks!
